pkg/agent/proxy/proxier_test.go

```diff
@@ -521,3 +522,76 @@ func TestSessionAffinityIPv4(t *testing.T) {
 func TestSessionAffinityIPv6(t *testing.T) {
 	testSessionAffinity(t, net.ParseIP("5060:70::81"), net.ParseIP("10:20::41"), true)
 }
+
+func testPortChange(t *testing.T, svcIP net.IP, epIP net.IP, isIPv6 bool) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockOFClient := ofmock.NewMockClient(ctrl)
+	fp := NewFakeProxier(mockOFClient, isIPv6)
+
+	svcPort1 := 80
+	svcPort2 := 8080
+	svcPortName := k8sproxy.ServicePortName{
+		NamespacedName: makeNamespaceName("ns1", "svc1"),
+		Port:           "http",
+		Protocol:       corev1.ProtocolTCP,
+	}
+	service := makeTestService(svcPortName.Namespace, svcPortName.Name, func(svc *corev1.Service) {
+		svc.Spec.ClusterIP = svcIP.String()
+		svc.Spec.Ports = []corev1.ServicePort{{
+			Name:       svcPortName.Port,
+			Port:       int32(svcPort1),
+			TargetPort: intstr.FromInt(80),
+			Protocol:   corev1.ProtocolTCP,
+		}}
+	})
+	makeServiceMap(fp, service)
+
+	epFunc := func(ept *corev1.Endpoints) {
+		ept.Subsets = []corev1.EndpointSubset{{
+			Addresses: []corev1.EndpointAddress{{
+				IP: epIP.String(),
+			}},
+			Ports: []corev1.EndpointPort{{
+				Name:     svcPortName.Port,
+				Port:     int32(80),
+				Protocol: corev1.ProtocolTCP,
+			}},
+		}}
+	}
+
+	bindingProtocol := binding.ProtocolTCP
+	if isIPv6 {
+		bindingProtocol = binding.ProtocolTCPv6
+	}
+	ep := makeTestEndpoints(svcPortName.Namespace, svcPortName.Name, epFunc)
+	makeEndpointsMap(fp, ep)
+	groupID, _ := fp.groupCounter.Get(svcPortName)
+	mockOFClient.EXPECT().InstallServiceGroup(groupID, false, gomock.Any()).Times(2)
+	mockOFClient.EXPECT().InstallEndpointFlows(bindingProtocol, gomock.Any(), isIPv6).Times(2)
+	mockOFClient.EXPECT().InstallServiceFlows(groupID, svcIP, uint16(svcPort1), bindingProtocol, uint16(0))
+	mockOFClient.EXPECT().UninstallServiceFlows(svcIP, uint16(svcPort1), bindingProtocol)
```

Comment:
should it be svcPort2?

Reply:
The flows for `svcPort1` are removed and the flows for `svcPort2` are installed to replace them, since the update in the Service spec is `svcPort1` -> `svcPort2`

Reply:
Sorry, my mistake, I was trying to find why https://github.com/vmware-tanzu/antrea/pull/1576/files/5e462a536346be015358140c42307167258b60b1#r526558336 was not detected in tests.

Reply:
I noticed that we don't really have tests for the LB flows...